lib/Sema/TypeCheckPattern.cpp

```diff
@@ -124,14 +188,110 @@ lookupUnqualifiedEnumMemberElement(TypeChecker &TC, DeclContext *DC,
 /// Find an enum element in an enum type.
 static EnumElementDecl *
 lookupEnumMemberElement(TypeChecker &TC, DeclContext *DC, Type ty,
-                        Identifier name, SourceLoc UseLoc) {
+                        DeclName name, SourceLoc UseLoc) {
   if (!ty->mayHaveMembers())
     return nullptr;
 
   // Look up the case inside the enum.
   // FIXME: We should be able to tell if this is a private lookup.
   NameLookupOptions lookupOptions = defaultMemberLookupOptions;
-  LookupResult foundElements = TC.lookupMember(DC, ty, name, lookupOptions);
+  LookupResult foundElements = TC.lookupMember(DC, ty, name,
+                                               lookupOptions);
+
+  ArrayRef<Identifier> queryParams = name.getArgumentNames();
+
+  if(foundElements.size() == 0){
+    // Look up enum element only with base name
+    foundElements = TC.lookupMember(DC, ty, name.getBaseIdentifier(),
+                                               lookupOptions);
+
+    // Prune results that don't match
+    LookupResult revisedResult;
+    for (LookupResultEntry resultEnt : foundElements) {
+      ValueDecl *e = resultEnt.getValueDecl();
+      // Maintaining part of list list for filter results
+      if (auto *var = dyn_cast<VarDecl>(e)) {
+        revisedResult.add(resultEnt, false);
+        continue;
+      }
+      DeclName enumElement = resultEnt.getValueDecl()->getFullName();
+      ArrayRef<Identifier> enumElementParams = enumElement.getArgumentNames();
+
+      bool enumHasLabels = false;
+      for(size_t i = 0;i<enumElementParams.size();++i)
+        enumHasLabels = enumHasLabels || !enumElementParams[i].empty();
+      if(foundElements.size() == 1 && !enumHasLabels){
+        revisedResult.add(resultEnt, false);
+        continue;
+      }
+      // If size is not the same, and query is not just a single value
+      // do not consider
+      if(enumElementParams.size() != queryParams.size() && 
+          queryParams.size() != 1){
+        continue;
+      }
+      bool addCandidate = true;
+      if(queryParams.size() == 1){
```

Comment:
This condition would always be true, right? Since we `continue` above if the size is not equal to 1.

Reply:
It isn't always true because it will continue if the sizes are not equal.  It should be `((enumElementParams.size() != queryParams.size()) &&
           (queryParams.size() != 1))` for clarity.